ignite/metrics/accuracy.py

```diff
@@ -152,7 +156,8 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
         if self._type == "binary":
             correct = torch.eq(y_pred.view(-1).to(y), y.view(-1))
         elif self._type == "multiclass":
-            indices = torch.argmax(y_pred, dim=1)
+            # indices = torch.argmax(y_pred, dim=1)
+            indices = self._logits_to_label_fn(y_pred)
```

Comment:
I would rather have such a function for all the cases (even if this is just `lambda x: x`) and apply them before the case switching to reduce the difference in behaviour among the cases

Reply:
IMHO, this is not a good idea. This reflects a design concern. I think that implementation of metrics should be splitted : (binary and multiclass) vs multilabel.
* binary is an specific case of multiclass. So we could apply the function in both cases.
* multilabel is different and should have a specific class.
Althought, a front-end class for user should be provided.